railties/lib/rails/application/configuration.rb

```diff
@@ -443,23 +443,12 @@ def colorize_logging=(val)
       #   config.session_store :my_custom_store
       def session_store(new_session_store = nil, **options)
         if new_session_store
-          if new_session_store == :active_record_store
-            begin
-              ActionDispatch::Session::ActiveRecordStore
-            rescue NameError
-              raise "`ActiveRecord::SessionStore` is extracted out of Rails into a gem. " \
-                "Please add `activerecord-session_store` to your Gemfile to use it."
-            end
-          end
-
```

Comment:
I'm on the fence about removing this.  Without it, the error would be just "uninitialized constant ActionDispatch::Session::ActiveRecordStore (NameError)", which is not nearly as helpful.

What about adding a helpful, general error message by rescuing `NameError` raised by the `ActionDispatch::Session.const_get` further below.  Roughly something like:

```
Could not resolve session store #{@session_store.inspect}.

#{@session_store.inspect} resolves to ActionDispatch::Session::#{@session_store.to_s.camelize}, but ActionDispatch::Session::#{@session_store.to_s.camelize} is not defined.

Is #{@session_store.inspect} spelled correctly, and are the necessary gems installed?
```

It might be possible to add some kind of did-you-mean suggestions based on the constants under `ActionDispatch::Session`, but that may be more trouble than it's worth.  (And could be a separate PR.)

Reply:
I like the idea for a DYM error, I'll take a stab at that in a follow up PR